modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -134,21 +196,131 @@ const MenuList = styled('ul')<Pick<SelectProps, 'error'>>(
+
+    let fallbackPlacements: Placement[] = [];
+    if (isAutoFlipped) {
+      fallbackPlacements = isFlipped ? ['bottom-start'] : ['top-start'];
+    }
+
+    const modifiers = [
+      {
+        name: 'flip',
+        options: {
+          fallbackPlacements,
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: ({
+            placement,
+            reference,
+            popper,
+          }: {
+            placement: Placement;
+            reference: Rect;
+            popper: Rect;
+          }) => {
+            // Skid menu along the edge of the button to account
+            // for fractional x-positioning of the button (e.g.,
+            // if the button is in a horizontally-centered modal)
+            // and to ensure proper alignment between the button
+            // and the menu
+            const skidding = reference.x % 1 === 0 ? 0 : 1;
+            // Displace menu towards the button to obscure the bottom
+            // edge of the button and to create a smooth visual
+            // connection between the button and the menu
+            const distance = -parseInt(borderRadius.m, 10);
+            return [skidding, distance];
+          },
+        },
+      },
+    ];
+
+    return {
+      modifiers,
+      onFirstUpdate: () => {
+        if (isAutoFocused && menuRef && menuRef.current) {
+          menuRef.current.focus();
+        }
+      },
+    };
+  };
+
+  private updateWidth = () => {
+    const {buttonRef} = this.props;
+    if (buttonRef.current) {
+      const newMenuWidth = buttonRef.current.clientWidth + 2 * buttonBorderWidth;
+      // Only update width state if it actually changed
+      if (newMenuWidth !== this.state.width) {
+        this.setState({width: newMenuWidth});
+      }
+    }
+  };
+
+  componentDidMount() {
+    this.updateWidth();
+
+    // Update menu width state on resize to ensure menu resizes as window resizes
+    window.addEventListener('resize', this.updateWidth);
+  }
 
-export default SelectMenu;
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateWidth);
+  }
+
+  public render() {
+    const {
+      buttonRef,
+      children,
+      error,
+      isAnimated,
+      isAutoFlipped,
+      isAutoFocused,
+      isFlipped,
+      isHidden,
+      isHiding,
+      menuRef,
+      ...elemProps
+    } = this.props;
+
+    const {width} = this.state;
+
+    // Render nothing if buttonRef.current hasn't been set
+    return buttonRef.current ? (
+      <Popper
+        placement={isFlipped ? 'top-start' : 'bottom-start'}
+        open={!isHidden}
+        anchorElement={buttonRef.current}
+        popperOptions={this.generatePopperOptions()}
+        // zIndex is necessary in order for the menu to be displayed
+        // properly if Select is placed within a CK Modal (necessary
+        // to override zIndex of 1 for the Modal)
+        style={{zIndex: 2}}
```

Comment:
Wanted to call out the `zIndex` here. See code comment for an explanation.

Reply:
We cannot hard-code without an option to override. `zIndex` 2 will work in a Canvas Kit-only application with no other `zIndex` overrides, but will not work in an application with other z index values. Modal allows the z-index to be overridden and we should also do that here. Until we have a Popup manager #182 that works globally across all apps, we need to allow z-index to be overridden.

Reply:
Removed the zIndex after merging in the stack manager.